pkg/ccloudv2/flink_gateway.go

```diff
@@ -37,13 +37,15 @@ func NewFlinkGatewayClient(url, userAgent string, unsafeTrace bool, authToken st
 			if len(via) >= 5 {
 				return fmt.Errorf("stopped after 5 redirects")
 			}
-			log.CliLogger.Debugf("Following redirect with authorization to %s", req.URL)
-			// Customize the redirect to add authorization header on 307 Redirect.
-			// This is required as the Location header returned by the gateway may not be an exact subdomain and Authorization
-			// header is copied only when the hostname is exactly the same or an exact subdomain of the hostname
-			// Ideally, we should check the status code returned in via[0].Response. However, via[0].Response is nil in underlying
-			// retryable http implementation
-			req.Header.Add("Authorization", "Bearer "+authToken)
+			if len(via) > 0 {
```

Comment:
`len(via)` will at least be 1 since the redirect happens after at least 1 request was made. The `if` check maybe redundant. lmk if my understanding is incorrect.

Reply:
Yeah I think you're right, I just left it there to be safe